runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -396,8 +396,18 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
 
         long size = streamAddressSpace.size();
         if (size == 0L) {
-            // The trim mark will allow to detect trimmed exceptions if the seeked address falls behind it
-            return streamAddressSpace.getTrimMark();
+            // We want to optimize and directly return when no updates are found in advance (i.e., no need to do remainingUpTo)
+            // but we need to consider the case when the 'syncUpTo' address falls below the trim mark, so we correctly
+            // report the TrimmedException allowing consumers to know that data is no longer available from this point.
+            // We could throw the TrimmedException directly or in this case let remainingUpTo discover it for us.
+            if (getCurrentGlobalPosition() <= streamAddressSpace.getTrimMark()) {
+                // If pointer is -1L we must return the known trim mark or the trimmed exception will be lost
+                return getCurrentGlobalPosition() == Address.NON_ADDRESS ? streamAddressSpace.getTrimMark() : getCurrentGlobalPosition();
```

Comment:
does getCurrentGlobalPointer() return the max global seen by the sequencer?  or is it tracking the max address for this stream known so far(before querying the sequencer)?

Reply:
It is tracking the position at which the pointer is currently on for this stream (not necessarily the max)... In the case of subscription API (where this call is coming from) it is up to the position where we want to start syncing or we have synced up to...